projects/ssr-tests/src/ssr-testing.spec.ts

```diff
@@ -1,54 +1,126 @@
-import * as Log from './log.utils';
-import * as ProxyServer from './proxy.utils';
-import * as Ssr from './ssr.utils';
+import { Server } from 'http';
+import { expectLogMessages } from './matchers/matchers';
+import * as LogUtils from './utils/log.utils';
+import * as ProxyUtils from './utils/proxy.utils';
+import * as SsrUtils from './utils/ssr.utils';
 
 const BACKEND_BASE_URL: string = process.env.CX_BASE_URL || '';
 
 describe('SSR E2E', () => {
-  let proxy: any;
+  let proxy: Server;
   const REQUEST_PATH = '/electronics-spa/en/USD/';
 
-  beforeEach(async () => {
-    Log.clearSsrLogFile();
-    await Ssr.startSsrServer();
+  beforeEach(() => {
+    LogUtils.clearSsrLogFile();
   });
 
   afterEach(async () => {
-    await proxy.close();
-    await Ssr.killSsrServer();
+    proxy.close();
+    await SsrUtils.killSsrServer();
   });
 
-  it('should receive success response with request', async () => {
-    proxy = await ProxyServer.startProxyServer({
-      target: BACKEND_BASE_URL,
-    });
-    const response: any = await ProxyServer.sendRequest(REQUEST_PATH);
-    expect(response.statusCode).toEqual(200);
-
-    // Rendering should not complete in the first request.
-    // App should fall back to csr.
-    Log.assertMessages([
-      `Rendering started (${REQUEST_PATH})`,
-      `Request is waiting for the SSR rendering to complete (${REQUEST_PATH})`,
-    ]);
-  });
+  describe('With SSR error handling', () => {
+    describe('Common behavior', () => {
+      beforeEach(async () => {
+        await SsrUtils.startSsrServer();
+      });
+
+      it('should receive success response with request', async () => {
+        proxy = await ProxyUtils.startBackendProxyServer({
+          target: BACKEND_BASE_URL,
+        });
+        const response: any =
+          await ProxyUtils.sendRequestToSsrServer(REQUEST_PATH);
```

Comment:
[MINOR]
It's very confusing we're using `ProxyUtils` for sending a request to Ssr server.
Shouldn't it be simply kinda "HttpUtils`?

Reply:
Are you suggesting only renaming `ProxyUtils` or also moving `sendRequestToSsrServer` outsde `proxy.utils.ts` to a separate file, e.g. `http.utils.ts`?

Reply:
I'd like to distinguish 3 entities:
- SSR server - that you can `start` (with config), `kill`
- Backend proxy - that you can `start` (with config`, and... IMHO we should also kill it after each test to avoid memory leaks in tests? ;) I guess so... - perhaps `close`ing both `httpProxy` server object and `http` server object
- TestClient - that can `sendRequest` (and get response)